upstairs/src/lib.rs

```diff
@@ -1617,15 +1630,15 @@ impl Buffer {
         assert_eq!(offset % self.block_size, 0);
         assert_eq!(response.data.len(), self.block_size);
         if !response.block_contexts.is_empty() {
-            self.data[offset..][..response.data.len()]
+            self.block_mut(offset / self.block_size)
                 .copy_from_slice(&response.data);
-            self.owned[offset..][..response.data.len()].fill(true);
+            self.owned[offset / self.block_size] = true;
```

Comment:
style nit: It seems like other block/owned manipulations set `owned` before `block`.  I think it'd be nice to keep that consistent across the different methods in Buffer just for the sake of readability.

Reply:
Alright, done in 0b24644d8678fcfcdac84d8f8d0cc9da9af0d146